tests/services/test_browser.py

```diff
@@ -70,6 +71,36 @@ class MyServiceListener(r.ServiceListener):
     zc.close()
 
 
+# This test uses asyncio because ServiceBrowser.cancel uses call_soon_threadsafe
+@pytest.mark.asyncio
+async def test_service_browser_cancel_context_manager():
+    """Test we can cancel a ServiceBrowser with it being used as a context manager."""
+
+    # instantiate a zeroconf instance
+    zc = Zeroconf(interfaces=['127.0.0.1'])
+    # start a browser
+    type_ = "_hap._tcp.local."
+
+    class MyServiceListener(r.ServiceListener):
+        pass
+
+    listener = MyServiceListener()
+
+    browser = r.ServiceBrowser(zc, type_, None, listener)
+
+    assert cast(bool, browser.done) is False
+
+    with browser:
+        pass
+
+    # ensure call_soon_threadsafe is run
+    await asyncio.sleep(0)
```

Comment:
Maybe use `asyncio.run_coroutine_threadsafe(asyncio.sleep(0), zc.loop).result()` instead to keep the test sync so the sync versions are used in the expected way